python/pyspark/sql/functions.py

```diff
@@ -11157,6 +11157,136 @@ def udf(
         return _create_py_udf(f=f, returnType=returnType, useArrow=useArrow)
 
 
+# ---------------  Conditional functions ------------------------
+
+
+@try_remote_functions
+def ifnull(col1: "ColumnOrName", col2: "ColumnOrName") -> Column:
```

Comment:
current `ifnull` use the same expression as `nvl`, let's discuss in https://github.com/apache/spark/pull/41516#discussion_r1225030166 first

Reply:
let's keep current implementation which directly invoke alias `ifnull`